mgnify_pipelines_toolkit/analysis/assembly/antismash_gff_builder.py

```diff
@@ -8,121 +23,115 @@
-                if 'gene_kind' in row['qualifiers'].keys():
-                    row_type = row['qualifiers']['gene_kind'][0]
-                    note_str += f";as_type={row_type}"
-
-                    row_functions = ','.join(row['qualifiers']['gene_functions']).replace(' ', '_').replace(':_', ':').replace(';_', ';')
-                    note_str += f";gene_functions={row_functions}"
-
-                else:
-                    note_str += f";as_type=other"
-                
-                note_dict[row_id] += note_str
-
+        # Extended CDS attributes 
         if 'antismash.detection.hmm_detection' in record['modules'].keys():
-            temp_row = record['modules']['antismash.detection.hmm_detection']['rule_results']['cds_by_protocluster']
-            if len(temp_row) > 0:
-                for row in temp_row[0][1]:
-                    if 'cds_name' in row.keys():
-                        row_id = row['cds_name']
-                        row_as_clusters = ','.join(list(row['definition_domains'].keys()))
-                        if row_id in note_dict.keys():
-                            note_dict[row_id] += f';as_gene_clusters={row_as_clusters}'
+            cds_by_protocluster = record['modules']['antismash.detection.hmm_detection']['rule_results']['cds_by_protocluster'] # imprive var name
+            if len(cds_by_protocluster) > 0:
+                for feature in cds_by_protocluster[0][1]:
+                    if 'cds_name' in feature.keys():
+                        locus_tag = feature['cds_name']
+                        as_clusters = ','.join(list(feature['definition_domains'].keys()))
+                        if locus_tag in attributes_dict.keys():
+                            attributes_dict[locus_tag].update({'as_gene_clusters':as_clusters})
 
         if 'antismash.detection.genefunctions' in record['modules'].keys():
             for tool in record['modules']['antismash.detection.genefunctions']['tools']:
-                note_str = ""
-                if tool['tool'] == 'smcogs' and len(tool['best_hits']) > 0:
-                    for best_hit in tool['best_hits']:
-                        hit_id = tool['best_hits'][best_hit]['hit_id'].split(':')[0]
-                        hit_desc = tool['best_hits'][best_hit]['hit_id'].split(':')[1].replace(' ', '_')
-                        score = tool['best_hits'][best_hit]['bitscore']
-                        eval = tool['best_hits'][best_hit]['evalue']
-
-                        if "as_notes" in note_str:
-                            note_str += f'smCOG%3A%20{hit_id}%A{hit_desc}%20%28Score%3A%20{score}%3B%20E-value%3A%20{eval}%29%3B'
-                        else:
-                            note_str += f';as_notes=smCOG%3A%20{hit_id}%A{hit_desc}%20%28Score%3A%20{score}%3B%20E-value%3A%20{eval}%29%3B'
-                        if best_hit in note_dict.keys():
-                            note_dict[best_hit] += note_str
+                if tool['tool'] == 'smcogs':
```

Comment:
Based on my test data, this is redundant with the 'gene_functions' attributes of CDS. But I'm not entirely sure

Reply:
This gives us the SMCOG ID as well, so it's mostly redundant but there is that one piece of info that makes it worth keeping IMO